packages/teleterm/src/services/tshd/mapUsageEvent.ts

```diff
@@ -0,0 +1,89 @@
+import { Timestamp } from 'google-protobuf/google/protobuf/timestamp_pb';
+
+import * as api from './v1/usage_events_pb';
+import * as prehogApi from './prehog/v1alpha/connect_pb';
+
+import * as types from './types';
+
+export function mapUsageEvent(event: types.UsageEventWithDate) {
+  return new api.ReportUsageEventRequest()
+    .setAuthClusterId(event.authClusterId)
+    .setPrehogEvent(mapPrehogBody(event.prehogEvent));
+}
+
+function mapPrehogBody(
+  prehogEvent: types.UsageEventWithDate['prehogEvent']
+): prehogApi.SubmitConnectEventRequest {
+  const prehogApiEvent = new prehogApi.SubmitConnectEventRequest()
```

Comment:
I think we could clean up the var names a little bit because right now we have `prehogApiEvent`, `prehogEvent`, `event` and `apiEvent` and it's a bit hard to understand what's going on. How do you feel about the following:

* `prehogApiEvent` could become `req` since that's what it is.
* `apiEvent` could become `reqEvent` to signal that it's going to be assigned to `req`.

That would leave us with `prehogEvent` and `event`. I think what's important here is to signal that

1. `mapUsageEvent` maps a plain JS object into a gRPC request object (we should probably add a comment about it).
2. The input object is created on our side rather than consumed from tshd.

So how about renaming `prehogEvent` to `plainReq`?

I'd probably even go as far as renaming `prehog_event` in `ReportUsageEventRequest` to `prehog_req` – it's not an event in itself, it's a request body for submitting the event.

Reply:
I'm always happy to adjust the naming, I had some troubles finding meaningful names. I'm only wondering if `plainReq` shouldn't be called `prehogReq`?

Reply:
> I'm always happy to adjust the naming, I had some troubles finding meaningful names. I'm only wondering if `plainReq` shouldn't be called `prehogReq`?

When I was reading this file I thought that the most important part is that one variable holds a plain JS object and the other holds the gRPC form so it seemed that focusing on this distinction was the way to go.

`prehogReq` doesn't help in this scenario IMHO as both `plainReq` and `req` are "prehog reqs", it's just that one is a gRPC class and the other is not.

Reply:
>When I was reading this file I thought that the most important part is that one variable holds a plain JS object 

Right, makes sense.